Backpack/src/main/java/net/skyscanner/backpack/button/BpkButton.kt

```diff
@@ -100,126 +60,32 @@ open class BpkButton : AppCompatButton {
-  var iconPosition: Int = BpkButton.END
-    set(value) {
-      if (value != field) {
-        field = value
-        setUpIfInitialized()
-      }
-    }
-
-  override fun setCompoundDrawablesWithIntrinsicBounds(@DrawableRes left: Int, @DrawableRes top: Int, @DrawableRes right: Int, @DrawableRes bottom: Int) {
-    super.setCompoundDrawablesWithIntrinsicBounds(left, top, right, bottom)
-    updatePadding()
-  }
-
-  override fun setCompoundDrawablesWithIntrinsicBounds(left: Drawable?, top: Drawable?, right: Drawable?, bottom: Drawable?) {
-    super.setCompoundDrawablesWithIntrinsicBounds(left, top, right, bottom)
-    updatePadding()
-  }
-
-  override fun setText(text: CharSequence, type: TextView.BufferType) {
-    super.setText(text, type)
-    updatePadding()
-  }
-
-  override fun onSizeChanged(w: Int, h: Int, oldw: Int, oldh: Int) {
-    super.onSizeChanged(w, h, oldw, oldh)
-    updatePadding(w)
-  }
-
-  override fun setPadding(left: Int, top: Int, right: Int, bottom: Int) {
-    super.setPadding(left, top, right, bottom)
-    originalStartPadding = left
-    originalEndPadding = right
-    updatePadding(false)
-  }
-
-  override fun setPaddingRelative(start: Int, top: Int, end: Int, bottom: Int) {
-    super.setPaddingRelative(start, top, end, bottom)
-    originalStartPadding = start
-    originalEndPadding = end
-    updatePadding()
-  }
-
-  override fun setEnabled(enabled: Boolean) {
-    // Even though we have a StateListDrawable, it is not enough just to rely on the state of
-    // the background for enabled/disabled change as we have text color and other properties
-    // changing incl the stroke of the background which means we need to change that as well.
-    super.setEnabled(enabled)
-    setUpIfInitialized()
-  }
-
   private fun initialize(attrs: AttributeSet?, defStyleAttr: Int) {
     val attr = context.theme.obtainStyledAttributes(attrs, R.styleable.BpkButton, defStyleAttr, 0)
     try {
       if (attr.hasValue(R.styleable.BpkButton_buttonType)) {
         initialType = Type.fromId(attr.getInt(R.styleable.BpkButton_buttonType, 0))
       }
 
-      iconPosition = attr.getInt(R.styleable.BpkButton_buttonIconPosition, END)
-
       buttonBackgroundColor = attr.getColor(R.styleable.BpkButton_buttonBackgroundColor, BpkTheme.getColor(context, type.bgColor))
-      buttonTextColor = attr.getColor(R.styleable.BpkButton_buttonTextColor, BpkTheme.getColor(context, type.textColor))
 
       buttonStrokeColor = attr.getColor(R.styleable.BpkButton_buttonStrokeColor, BpkTheme.getColor(context, type.strokeColor))
       roundedButtonCorner = attr.getDimension(R.styleable.BpkButton_buttonCornerRadius, context.resources.getDimension(R.dimen.bpkSpacingLg))
 
-      attr.getResourceId(R.styleable.BpkButton_buttonIcon, INVALID_RESOURCE).let {
-        if (it != INVALID_RESOURCE) {
-          icon = AppCompatResources.getDrawable(context, it)
-        }
-      }
+      buttonTextColor = ContextCompat.getColor(context, type.textColor)
```

Comment:
Why we're not using attr.getColor(R.styleable.BpkButton_buttonTextColor) anymore?

Reply:
That's in the base class now as `overrideColour`. This is only set via theming

Reply:
I'd rename the attribute then to defaultColour and buttonTextColor. The current naming is confusing.

Reply:
done